presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -277,12 +281,21 @@ public Expression rewriteLambdaExpression(LambdaExpression node, Scope context,
             {
                 checkState(analysis.getCoercion(node) == null, "cannot coerce a lambda expression");
 
+                ImmutableList.Builder<Field> fields = ImmutableList.builder();
                 ImmutableList.Builder<LambdaArgumentDeclaration> newArguments = ImmutableList.builder();
-                for (LambdaArgumentDeclaration argument : node.getArguments()) {
+                for (int i = 0; i < node.getArguments().size(); i++) {
+                    LambdaArgumentDeclaration argument = node.getArguments().get(i);
+                    fields.add(Field.newUnqualified(argument.getName().getValue(), analysis.getType(argument)));
                     Symbol symbol = lambdaDeclarationToSymbolMap.get(NodeRef.of(argument));
                     newArguments.add(new LambdaArgumentDeclaration(new Identifier(symbol.getName())));
                 }
-                Expression rewrittenBody = treeRewriter.rewrite(node.getBody(), context);
+
+                Scope lambdaScope = Scope.builder()
```

Comment:
I am not a great fan of this change. This only adds complexity and more dependency on scopes. A simpler approach would be to just make existing resolution methods in `ExpressionAnalyzer` and `TranslationMap` consistent or fix resolution here to behave like scope resolution (and add couple of `TODOs`). See comment: https://github.com/prestodb/presto/pull/9026#issuecomment-333950469

`Scopes` are created as part of analysis, yet here you create a new scope (presumably compatible with the one from `ExpressionAnalyzer`) as part of planning. Thus this change brings more analyzer code to planner.

Reply:
@sopel39 this code already uses `Scope` (so the dependency is already here), but incorrectly. See, you cannot use scopes just a little, either you do correctly, with all what it takes, or you don't at all.

I am fixing a bug here (improper, too simplistic, use of scopes leading to wrong query results). I agree making planner removed from scopes is the better way to solve the problem, it is, however, also more complex (I tried but without success so far). Also, this PR doesn't make removing scopes from planning any harder.

Reply:
The code here uses scopes but only for relation columns and not for lambda arguments. You added completely new use case of `Scopes` (for resolving lambda references) and I think creating new scope explicitly here is kind of hacky way of doing things because it brings even more analyzer code to the planner. In any case, as bad as the old lambda resolution code is it is already there and I think it just required small fixed in order to close the mentioned issues. Please let not bring additional, unnecessary complexity here.

Reply:
In short, in current `TranslationMap` I don't see a proper (elegant) way of fitting lambda `Scopes`. `Scopes` are passed to `TranslationMap` externally (in `RewriteBase`) and should not be created explicitly here. The order of changes in my option should be:
1. fix lambda argument resolution mechanism in `ExpressionAnalyzer` and `TranslationMap` (preferably without introducing `Scope` for lambda arguments resolution in `ExpressionAnalzer`).
2. Complete https://github.com/prestodb/presto/issues/7398
3. Switch to `Scopes` for lambda argument resolution.